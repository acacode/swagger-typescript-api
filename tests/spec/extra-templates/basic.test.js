import * as fs from "node:fs/promises";
import * as os from "node:os";
import * as path from "node:path";

import { afterAll, beforeAll, describe, expect, test } from "vitest";

import { generateApi } from "../../../src/index.js";

describe("basic", async () => {
  let tmpdir = "";

  beforeAll(async () => {
    tmpdir = await fs.mkdtemp(path.join(os.tmpdir(), "swagger-typescript-api"));
  });

  afterAll(async () => {
    await fs.rm(tmpdir, { recursive: true });
  });

  test("--extra-templates", async () => {
    await generateApi({
      name: "schema",
      input: path.resolve(import.meta.dirname, "schema.json"),
      output: tmpdir,
      silent: true,
      extraTemplates: [
        {
          name: "external-template-name",
          path: path.resolve(import.meta.dirname, "./templates/test.ejs")
        },
      ],
    });

    const content = await fs.readFile(path.join(tmpdir, "external-template-name.ts"), {
      encoding: "utf8",
    });

    expect(content).toMatchSnapshot();
  });

  test("--extra-templates-route", async () => {
    await generateApi({
      name: "schema",
      input: path.resolve(import.meta.dirname, "schema.json"),
      output: tmpdir,
      silent: true,
      extraTemplates: [
        {
          name: (config)=>{
            console.log(config.route.moduleName)
            return `contexts/${config.route.moduleName}Context`
          },
          path: path.resolve(import.meta.dirname, "./templates/test-route.ejs"),
          modular: true
        },
      ],
    });

    const content = await fs.readFile(path.join(tmpdir, "contexts/wrongPathParams1Context.ts"), {
      encoding: "utf8",
    });

    expect(content).toMatchSnapshot();
  });
});
