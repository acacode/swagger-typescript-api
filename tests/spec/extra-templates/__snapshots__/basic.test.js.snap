// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic > --extra-templates 1`] = `
"/* eslint-disable */
/* tslint:disable */
/*
 * ---------------------------------------------------------------
 * ## THIS FILE WAS GENERATED VIA SWAGGER-TYPESCRIPT-API        ##
 * ##                                                           ##
 * ## AUTHOR: acacode                                           ##
 * ## SOURCE: https://github.com/acacode/swagger-typescript-api ##
 * ---------------------------------------------------------------
 */

/**
  Swagger Petstore
  1.0.0
  A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification
*/
export class Api {
  private request(params: Record<string, any>) {
    return fetch(params as any);
  }
}
"
`;

exports[`basic > --extra-templates-route 1`] = `
"/* eslint-disable */
/* tslint:disable */
/*
 * ---------------------------------------------------------------
 * ## THIS FILE WAS GENERATED VIA SWAGGER-TYPESCRIPT-API        ##
 * ##                                                           ##
 * ## AUTHOR: acacode                                           ##
 * ## SOURCE: https://github.com/acacode/swagger-typescript-api ##
 * ---------------------------------------------------------------
 */

/**
  Swagger Petstore
  1.0.0
  A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification
*/
export class WrongPathParams1 {
  private request(params: Record<string, any>) {
    return fetch(params as any);
  }
}
"
`;
